Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -258,17 +258,54 @@ class SmartPage_Context
     }
   };
 
-  public SmartPage_PageRequestManager_endRequest = function (sender: unknown, args: Sys.WebForms.EndRequestEventArgs)
+  public SmartPage_PageRequestManager_endRequest = (sender: unknown, args: Sys.WebForms.EndRequestEventArgs) =>
   {
-    if (args.get_error() != undefined && (args.get_error() as any).httpStatusCode == '500')
+    if (args.get_error())
     {
-      const errorMessage = args.get_error().message;
-      args.set_errorHandled(true);
+      let statusCode = (args.get_error() as any).httpStatusCode;
 
-      const errorBody = '<div class="SmartPageErrorBody"><div>' + errorMessage + '</div></div>';
+      if (!statusCode || statusCode === 200)
+      {
+        this.HandleRequestError("undefined", args);
+      }
+      else
+      {
+        this.HandleRequestError(statusCode, args);
+      }
+    }
+  }
+
+  private HandleRequestError(statusCode: string, args: Sys.WebForms.EndRequestEventArgs): void
+  {
+    let response = args.get_response();
+    let errorMessage = args.get_error().message;
 
-      SmartPage_Context.Instance!.ShowMessage("SmartPageServerErrorMessage", errorBody);
+    //@ts-ignore
+    let serverError: string = Sys.WebForms.Res.PRM_ServerError;
+    let isSynchronousError = errorMessage.includes(serverError.substring(0, serverError.indexOf(':')));
+
+    let errorBody = '<div class="SmartPageErrorBody"><div>';
+
+    const shortenedErrorMessage = errorMessage.substring(errorMessage.indexOf(':') + 1);
+    errorBody += shortenedErrorMessage;
+
+    if (isSynchronousError)
+    {
+      errorBody += ', ' + response.get_statusText() + '<br>';
```

Comment:
```suggestion
      errorBody += ', ' + response.get_statusText() + '\n';
```
We don't need a BR because iframe is already a blocklevel element. Adding a newline is still helpful for reading etc.